whois-nrtm4/src/main/java/net/ripe/db/nrtm4/dao/SourceRepository.java

```diff
@@ -0,0 +1,74 @@
+package net.ripe.db.nrtm4.dao;
+
+import net.ripe.db.nrtm4.domain.NrtmSourceModel;
+import net.ripe.db.whois.common.domain.CIString;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.stereotype.Repository;
+
+import javax.sql.DataSource;
+import java.util.List;
+import java.util.Optional;
+
+
+@Repository
+public class SourceRepository {
+
+    private final JdbcTemplate jdbcTemplate;
+    private final List<String> sources;
+    private final String source;
+
+    SourceRepository(
+        @Qualifier("nrtmDataSource") final DataSource dataSource,
+        @Value("${whois.source}") final String source,
+        @Value("${whois.nonauth.source}") final String nonauthSource
+    ) {
+        this.jdbcTemplate = new JdbcTemplate(dataSource);
+        this.source = source.toUpperCase();
+        sources = List.of(
+            source.toUpperCase(),
+            nonauthSource.toUpperCase()
+        );
+    }
+
+    public void createSources() {
+        final String sql = """
+            INSERT INTO source (name)
+            VALUES (?)
+            """;
+        for (final String source : sources) {
+            jdbcTemplate.update(sql, source);
+        }
+    }
+
+    public Optional<NrtmSourceModel> getSource() {
+        final String sql = """
```

Comment:
May be rename to getMainSource or getWhoisSource as by default we are sending whois source?

Reply:
only used by tests so better to remove it from production code. renamed for now.